### PR TITLE
docs: Update changelog for release 3.1

### PR DESCRIPTION
**Pull request by lenny-intel**
If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-ui-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

**Comment by lenny-intel** [comment]
recheck

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,42 @@
...
+- Remove snap packaging ([6eeb2ce…](https://github.com/edgexfoundry/edgex-ui-go/commit/6eeb2ce6fc2afc5ffacdf29fd0663685f6ab0964))
+```text
+
+BREAKING CHANGE: Remove snap packaging
+
+```
+- New Provision Watchers tab
+  - Add Provision Watchers tab to Metadata ([a321165…](https://github.com/edgexfoundry/edgex-ui-go/commit/a321165113bd0bfa455e6042dab80b5a852e9ebb))
+  - Fields for Provision Watcher Tab DTO ([f7e57eb…](https://github.com/edgexfoundry/edgex-ui-go/commit/f7e57ebd2ad4d96da9d70341491ca8456df46195))
+  - More for Add Provision Watchers tab to Metadata ([e723d52…](https://github.com/edgexfoundry/edgex-ui-go/commit/e723d524dc2486b7380e56c06e54ac33686a11d2))
```
Comment (ejlee3) [inline CHANGELOG.md:28]: Is this the right wording? or should it be something else? It's a little awkward as is.

```suggestion
  - Add Metadata for Add Provision Watchers tab ([e723d52…](https://github.com/edgexfoundry/edgex-ui-go/commit/e723d524dc2486b7380e56c06e54ac33686a11d2))
```

**Review by ejlee3 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (lenny-intel) [reply CHANGELOG.md:28]: I changed is so it wasn't duplicate of first one. How about `Additional Add Provision Watchers tab to Metadata`?

Comment (ejlee3) [reply CHANGELOG.md:28]: Additional changes for Add...

**Review by ejlee3 — CHANGES_REQUESTED** [review]

**Review by ejlee3 — APPROVED** [review]
LGTM

**Comment by lenny-intel** [comment]
recheck

**Comment by lenny-intel** [comment]
@cloudxxx8 , I need you `committer` approval on this PR.

**Comment by lenny-intel** [comment]
recheck

**Comment by lenny-intel** [comment]
recheck

**Comment by lenny-intel** [comment]
recheck

**Review by ernestojeda — APPROVED** [review]

**Pull request merged**